dashboard/modules/reporter/reporter_head.py

```diff
@@ -411,6 +411,103 @@ async def get_task_cpu_profile(self, req) -> aiohttp.web.Response:
             headers={"Content-Type": "text/html"},
         )
 
+    @routes.get("/task/memory_profile")
+    async def task_memory_profile(self, req) -> aiohttp.web.Response:
+        """
+        Retrieves the memory profile for a specific task.
+        Note that one worker process works on one task at a time
+        or one worker works on multiple async tasks.
+
+        Args:
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the memory profile data.
+
+        Raises:
+            ValueError: If the "task_id" parameter is
+            missing in the request query.
+            ValueError: If the "attempt_number" parameter is
+            missing in the request query.
+            ValueError: If the maximum duration allowed is exceeded.
+            ValueError: If the worker begins working on
+            another task during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is
+            an internal server error during the profile retrieval.
+        """
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
```

Comment:
For my understanding, what is `attempt_number` and why is it important?

Reply:
So `attempt_number` in task represent the number of retries allowed to a failed task where for each attempt, the task can be run on different worker => different pid

Reply:
Oh I see, so a `(task_id, attempt_number)` pair uniquely identifies the process to track?

I'm a bit confused why the `task_id` isn't enough. I assume a task can only have 1 attempt running at a time, and memray only works on running processes. Couldn't we use that fact to translate `task_id` directly to the ip address and pid?

Reply:
yes, if we have the task id, most likely the last attempt is the one that has the actual worker that run the task (which the information is stored under `StateApi`). Otherwise, we might have like `(task_id, attempt_number-1)` which has worker running for different task (since the worker of failed task can be reused to run different task).